nncf/experimental/torch/sparsity/movement/structured_mask_handler.py

```diff
@@ -169,14 +169,31 @@ def update_independent_structured_mask_from_operand(self):
         self.independent_structured_mask = structured_mask
         return structured_mask
 
+    def initialize_binary_mask(self):
+        """
+        Initialize binary mask by all ones. The inflated dependent mask will be applied via logical "and"
+        operation to it. It's needed for the case when 1 binary mask shared for 2 groups: in one group operator
+        can be pruned by input channels, i.e. be a consumer of pruning masks, and for another - can be pruned by
+        output channels, i.e. be a producer of pruning masks.
+        Initial  |  Mask 2 last input channels   |  Mask middle output channel    |     Result
+        --------------------------------------------------------------------------------------
+         1111                 1100                             1111                     1100
+         1111    &            1100               &             0000                =    0000
+         1111                 1100                             1111                     1100
+        """
+
+        self.sparsifier_operand.weight_ctx.binary_mask.fill_(1)
+        if self.sparsifier_operand.prune_bias:
+            self.sparsifier_operand.bias_ctx.binary_mask.fill_(1)
+
     def populate_dependent_structured_mask_to_operand(self):
         """
         Updates the actual binary masks in operand with `self.dependent_structured_mask`.
         """
         structured_mask_inflated = self._inflate_structured_mask(self.dependent_structured_mask, self.grid_size)
-        self.sparsifier_operand.weight_ctx.binary_mask = structured_mask_inflated
+        self.sparsifier_operand.weight_ctx.binary_mask *= structured_mask_inflated
```

Comment:
@vuiseng9 @yujiepan-work 
Assume that controller initializes binary mask by ones. If some linear layer is pruned by row and column at the same time, the corresponding sparsifier operand will update binary 2 times: first multiply ones with the mask per row, second – multiply the produced mask with per column. I've verified it with 2 tests: 
[test_populate_dependent_structured_mask_by_row_and_column](https://github.com/openvinotoolkit/nncf/pull/1724/files#diff-2c85b01343c47e2fab4bcc0c8b0f78549343b2ff261cdc7c124660d790cce0d2R233)
[the ngraph transformation test for MobileBERT's recipe with all layers sparsified by default in fine-grained mode](https://github.com/openvinotoolkit/nncf/pull/1724/files#diff-65c84e2828d4233ff8ba02f754b7a5a644a922006cae2ad0346529e96ae11934R189)